### PR TITLE
Add unit test crate for stylo

### PR DESCRIPTION
**Pull request by Manishearth**
@canaltinova needs this for his mask work -- we need a way to test stylo code (or style code that's conditionally compiled out for gecko)

We can add this to the CI as well.

r? @emilio

---

This change is [Reviewable](https://reviewable.io/reviews/servo/servo/13335)

### CODE REVIEW

**Comment by emilio** [comment]
Can you reuse, or move out the `ports/geckolib/sanity-checks.rs` file?

**Comment by Manishearth** [comment]
Done.

File: tests/unit/stylo/lib.rs [hunk 1]
```diff
@@ -0,0 +1,13 @@
+extern crate app_units;
+extern crate cssparser;
+extern crate gecko_bindings;
+extern crate style;
+extern crate style_traits;
+
+mod sanity_checks;
+
+#[test]
+fn test_test() {
```
Comment (emilio) [inline tests/unit/stylo/lib.rs]: You can remove this now we have other tests, btw.

r=me with that, thanks! :)

File: python/servo/testing_commands.py [hunk 2]
```diff
@@ -232,6 +234,23 @@ def test_unit(self, test_name=None, package=None):
...
+        self.ensure_bootstrapped()
+
+        env = self.build_env()
+        env["RUST_BACKTRACE"] = "1"
+        env["CARGO_TARGET_DIR"] = path.join(self.context.topdir, "target", "geckolib").encode("UTF-8")
+
+        with cd(path.join("ports", "geckolib")):
+            result = call(["cargo", "test", "-p", "stylo_tests"], env=env)
+
+        if result != 0:
```
Comment (aneeshusa) [inline python/servo/testing_commands.py]: Why not always `return result` instead of hiding it behind the conditional?

**Comment by aneeshusa** [comment]
Please make a PR or Issue at saltfs if you want this to run on the CI.

File: python/servo/testing_commands.py (hunk 2 above)
Comment (Manishearth) [inline python/servo/testing_commands.py]: That's what the other commands do. I think mach has some special behavior here.

**Comment by Manishearth** [comment]
Yep, will do!

File: python/servo/testing_commands.py (hunk 2 above)
Comment (Wafflespeanut) [inline python/servo/testing_commands.py]: Looks like that should be cleaned up :smile:

**Pull request merged**